packages/vesting-base-lp/src/handlers.rs

```diff
@@ -0,0 +1,779 @@
+        .amount;
+    let paired_asset_balance_after_withdrawal = deps
+        .querier
+        .query_balance(env.contract.address.to_string(), paired_asset_denom.clone())?
+        .amount;
+
+    // calc amount of assets that's been withdrawn
+    let withdrawn_ntrn_amount = ntrn_balance_after_withdrawal.checked_sub(ntrn_init_balance)?;
+    let withdrawn_paired_asset_amount =
+        paired_asset_balance_after_withdrawal.checked_sub(paired_asset_init_balance)?;
+
+    let mut msgs: Vec<CosmosMsg> = vec![];
+
+    if !withdrawn_ntrn_amount.is_zero() && !withdrawn_paired_asset_amount.is_zero() {
+        msgs.push(CosmosMsg::Wasm(WasmMsg::Execute {
+            contract_addr: cl_pair_address.to_string(),
+            msg: to_binary(&PairExecuteMsg::ProvideLiquidity {
+                assets: vec![
+                    native_asset(ntrn_denom.clone(), withdrawn_ntrn_amount),
+                    native_asset(paired_asset_denom.clone(), withdrawn_paired_asset_amount),
+                ],
+                slippage_tolerance: Some(slippage_tolerance),
+                auto_stake: None,
+                receiver: None,
+            })?,
+            funds: vec![
+                Coin::new(withdrawn_ntrn_amount.into(), ntrn_denom),
+                Coin::new(withdrawn_paired_asset_amount.into(), paired_asset_denom),
+            ],
+        }))
+    }
+
+    msgs.push(CallbackMsg::PostMigrationVestingReschedule { user }.to_cosmos_msg(&env)?);
+
+    Ok(Response::default().add_messages(msgs))
+}
+
+fn post_migration_vesting_reschedule_callback(
+    deps: DepsMut,
+    env: Env,
+    user: &VestingAccountResponse,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    let migration_config: XykToClMigrationConfig = XYK_TO_CL_MIGRATION_CONFIG.load(deps.storage)?;
+    let balance_response: BalanceResponse = deps.querier.query_wasm_smart(
+        &migration_config.new_lp_token,
+        &Cw20QueryMsg::Balance {
+            address: env.contract.address.to_string(),
+        },
+    )?;
+    let current_balance = balance_response.balance;
+
+    let schedule = user.info.schedules.last().unwrap();
+
+    let new_end_point;
+    if let Some(end_point) = &schedule.end_point {
+        new_end_point = Option::from(vesting_base_pcl::types::VestingSchedulePoint {
+            time: end_point.time,
+            amount: current_balance,
+        })
+    } else {
+        new_end_point = None
+    }
+
+    let new_schedule = vesting_base_pcl::types::VestingSchedule {
+        start_point: vesting_base_pcl::types::VestingSchedulePoint {
+            time: schedule.start_point.time,
+            amount: Uint128::zero(),
+        },
+        end_point: new_end_point,
+    };
+
+    let vesting_info = vesting_info(config.extensions.historical);
+
+    vesting_info.save(
+        deps.storage,
+        user.address.clone(),
+        &VestingInfo {
+            schedules: vec![],
+            released_amount: Uint128::zero(),
```

Comment:
you save it as zero, i.e. remove vested tokens from contract's state, but here https://github.com/neutron-org/neutron-tge-contracts/blob/fbe0792432cbe22b9ad5bfea00fdb3a8680da317/packages/vesting-base-lp/src/handlers.rs#L452 you track the migrated amount in the total released amount. math wouldn't add up: total amount increases, but the eventual sum of all vesting infos after migration will be zero